lib/rouletteManager.js

```diff
@@ -43,11 +43,27 @@ RouletteManager.prototype.start = function (duration, price, onEnd) {
         if (this.listedUsers.length > 0) {
             var queueLength = this.BOT.getQueue().length;
 
-            for (var tries = 0; tries < 10 && oldSpot === newSpot; tries++) {
-                newSpot = Math.dice(queueLength * .8);
-                if (newSpot + queueLength * .2 > oldSpot) {
-                    newSpot += Math.floor(queueLength * .2);
-                }
+            var Switch = Math.dice(3);
+
+            // If they join while position one, then they can be subject to the old algorithm.
+            if (oldSpot = '1')
+                Switch = 3;
+
+            // Give a 2/3rds chance to move up, and 1/3rds chance for the old 'high risk' roll
+            switch (Switch) { 
```

Comment:
Why are you trying to switch over a string when `var Switch = Math.dice(3);` returns a number?
None of the switch cases will work.